src/html-purify.js

```diff
@@ -53,14 +53,56 @@ See the accompanying LICENSE file for terms.
         return false;
     }
 
+    function addQuoteToValue(action, value) {
+        if (action & derivedState.TransitionName.DQ_ATTR) {         // collect double quoted value
+            return '"' + value + '"';
+        } 
+        else if (action & derivedState.TransitionName.SQ_ATTR) {    // collect single quoted value
+            return "'" + value + "'";
+        } 
+        else /*if (action & derivedState.TransitionName.UQ_ATTR) {  // collect unquoted value */{
+            return value;
+        }
+    }
+
     function processTransition(prevState, nextState, i) {
         /* jshint validthis: true */
         /* jshint expr: true */
         var parser = this.parser,
+            action = derivedState.Transitions[prevState][nextState],
             idx, tagName, attrValString, openedTag, key, value;
 
-        
-        switch (derivedState.Transitions[prevState][nextState]) {
+        // check if tag and/or attr is available for collection
+        // collect the values (if any) only when the attr name is allowed
+        if (action & 0xF && contains(this.attributesWhitelist, (key = parser.getAttributeName()))) {
```

Comment:
we're doing attribute processing regardless of whether the tag is whitelisted or not, correct? can that be changed?